src/main/scala/com/trueconnectivity/e2e/utils/HealthCheck.scala

```diff
@@ -0,0 +1,41 @@
+package com.trueconnectivity.e2e.utils
+
+import scalaj.http.{Http, HttpResponse}
+
+import scala.util.{Failure, Success, Try}
+
+object HealthCheck {
+  def waitUntilHealthy(applicationPort: Int, retryCount: Int, retryDelay: Int): Unit = {
+    retry(retryCount, retryDelay) {
+      Http(s"http://localhost:${applicationPort}/health/status")
```

Comment:
i suggest we explore the possibility to address the docker container directly without exposing the port on localhost, so that it could be run without interfering with other apps

Reply:
I think that would involve running the tests inside Docker which may well be possible but would probably make debugging them much harder. Port clashes are a downside of running thing in Docker though so if you cn think of a way round it that would be good.

Reply:
you can just replace `localhost` here with the ip of the docker container and it should be ok, unless the docker-container network is isolated from host (which should not be the case by default)

Reply:
How do you get hold of the DockerContainer IP when the docker-compose network starts?

Reply:
`docker inspect "$1" | jq -r .[0].NetworkSettings.Networks[].IPAddress` where `$1` is the identificator of the container

Reply:
You would then also need to get the container id for the name of the service inside docker-compose to tie things together wouldn't you? At which point it is getting quite complicated...

Reply:
One step at a time. If we would need it we will as per my comment.